Allow cross zone creation of keyvault cert requests from dns registra…

hantonelli opened the pull request:
This PR allows to decouple the creation of keyvault cert request from the dns domain name registration resource blocks

arnaudlh commented:
hi @hantonelli thanks for the PR.

when trying to run example 103, I get the following error:

```
Error: Error in function call
│ 
│   on ../modules/security/keyvault_certificate_request/module.tf line 35, in resource "azurerm_key_vault_certificate" "csr":
│   35:         subject            = try(x509_certificate_properties.value.subject, "CN=${try("${x509_certificate_properties.value.domain_name_registration.subdomain}.", "")}${var.domain_name_registrations[try(x509_certificate_properties.value.domain_name_registration.lz_key, var.client_config.landingzone_key)][x509_certificate_properties.value.domain_name_registration.key].dns_domain_registration_name}")
│     ├────────────────
│     │ var.client_config.landingzone_key is "examples"
│     │ var.domain_name_registrations is object with 1 attribute "examples"
│     │ x509_certificate_properties.value is object with 4 attributes
│     │ x509_certificate_properties.value.domain_name_registration.key is "domain1"
│     │ x509_certificate_properties.value.domain_name_registration.lz_key is "example-domain-lz-key"
│     │ x509_certificate_properties.value.domain_name_registration.subdomain is "*"
│ 
│ Call to function "try" failed: no expression succeeded:
│ - Unsupported attribute (at ../modules/security/keyvault_certificate_request/module.tf:35,67-75)
│   This object does not have an attribute named "subject".
│ - Invalid index (at ../modules/security/keyvault_certificate_request/module.tf:35,198-305)
│   The given key does not identify an element in this collection value.
│ 
│ At least one expression must produce a successful result.
╵
Terraform plan return code: 1
Error on or near line 391: Error running terraform plan; exiting with status 1
```

hantonelli commented:
Hi @arnaudlh, I updated the example. It show how to create a keyvault_certificate_requests using a domain_name_registration  created in the same landing zone. The example has the lz_key commented out as the examples are designed to run on a single landing zone. I had added a README.md with relevant comments, I hope that helps.

hantonelli commented:
Hi @arnaudlh, I updated the example and added a README.md . The example show how to create a keyvault_certificate_requests using a domain_name_registrations, created in the same landing zone. The example has the  lz_key property commented out, as the examples are designed to run in a single landing zone, but is an example of how to use that property. I hope that the update example and new README.md help. Thank you for testing this PR.
